cattle.go

```diff
@@ -25,13 +25,13 @@ func NewCattleClient(cattleUrl string, accessKey string, secretKey string) (*Cat
 	}, nil
 }
 
-func (c *CattleClient) UpdateServiceDomainName(serviceDnsRecord utils.ServiceDnsRecord) error {
+func (c *CattleClient) UpdateServiceDomainName(metadataRecord utils.MetadataDnsRecord) error {
 	event := &rancher.ExternalDnsEvent{
 		EventType:   "dns.update",
-		ExternalId:  serviceDnsRecord.Fqdn,
-		ServiceName: serviceDnsRecord.ServiceName,
-		StackName:   serviceDnsRecord.StackName,
-		Fqdn:        serviceDnsRecord.Fqdn,
+		ExternalId:  metadataRecord.DnsRecord.Fqdn,
+		ServiceName: metadataRecord.ServiceName,
+		StackName:   metadataRecord.StackName,
+		Fqdn:        utils.UnFqdn(metadataRecord.DnsRecord.Fqdn),
```

Comment:
@prachidamle Template must result in a DNS name that conforms to RFC 1123, that is the template must consist of only alphanumeric characters, dashes ("-") as well as one or more of the supported placeholders.
Example of a valid template: `%{{service_name}}-%{{stack_name}}.customlabel`
Given root domain is `example.com` the resulting FQDN for stackFoo/serviceBar will be:
`servicebar-stackfoo.customlabel.example.com`